Objects/obmalloc.c

```diff
@@ -103,6 +109,39 @@ _PyMem_RawFree(void *ctx, void *ptr)
     free(ptr);
 }
 
+static void *
+_PyMem_RawAlignedAlloc(void *ctx, size_t alignment, size_t size)
+{
+    if (size == 0) {
+        size = 1;
+    }
+
+#ifdef MS_WINDOWS
+    return _aligned_malloc(size, alignment);
+#else
+    void *ptr;
+
+    if (posix_memalign(&ptr, alignment, size) < 0) {
+        return NULL;
+    }
+    if (ptr == NULL) {
```

Comment:
This looks like an accidental leftover from an earlier version. :)

Reply:
The manual page doesn't explicitly say that posix_memalign() never returns NULL on success. Whereas NULL is documented as an error condition for aligned_alloc(), memalign(), valloc() et pvalloc(). Why does posix_memalign() return an integer if it cannot return NULL on success? In case of doubt, I chose to explicitly reject NULL.

Reply:
At least, size=0 is replaced with size=1, as do in other Python memory allocators, to avoid returning NULL. posix_memalign() manual page:
"If size is 0, then the value placed in *memptr is either NULL, or a unique pointer value that can later be successfully passed to free(3)."

Reply:
http://pubs.opengroup.org/onlinepubs/009695399/functions/posix_memalign.html is clearer:

"Upon successful completion, posix_memalign() shall return zero; otherwise, an error number shall be returned to indicate the error."

> Why does posix_memalign() return an integer if it cannot return NULL on success? 

Someone just thought it would be a good idea to return ENOMEM etc. directly. Perhaps
just to avoid the `errno` dance.

The opengroup wording sounds like "success if and only if the return value is `0`.

Reply:
Fixed: I replaced the test with an assertion ;-)